internal/gps/vcs_source.go

```diff
@@ -316,23 +316,35 @@ func (s *gopkginSource) listVersions(ctx context.Context) ([]PairedVersion, erro
 	k := 0
 	var dbranch int // index of branch to be marked default
 	var bsv semver.Version
+	var master PairedVersion
+	tryMasterAsV0 := s.major == 0
 	for _, v := range ovlist {
 		// all git versions will always be paired
 		pv := v.(versionPair)
 		switch tv := pv.v.(type) {
 		case semVersion:
+			tryMasterAsV0 = false
 			if tv.sv.Major() == s.major && !s.unstable {
 				vlist[k] = v
 				k++
 			}
 		case branchVersion:
+			if tv.name == "master" {
```

Comment:
this binds us to the literal branch name `master`, which isn't guaranteed to be right. despite the gopkg.in docs on their site, it's also not actually what they do:

https://github.com/niemeyer/gopkg/blob/c70d35213a11190a3f9d921a421d4ec04620e60f/main.go#L410-L418

we _may_ be able to do something less complicated than what we have in `gitSource.listVersions()`, but we have to rely on the `HEAD` value in the output list.

Reply:
Thanks for catching that. They saying `master` in the docs, and in the code comments, and in some hard-coded strings so I thought that's what they were using.